iosxr_iso2vbox.py

```diff
@@ -65,38 +63,299 @@
+    odd double return on vbox
+    '''
+    logger.info('Logging into Vagrant Virtualbox and configuring IOS XR')
+
+    localhost = 'localhost'
+    prompt = r"[$#]"
+
+    def xr_cli_wait_for_output(command, pattern):
+        '''
+        Execute a XR CLI command and try to find a pattern.
+        Try up to five times them register an error.
+        Each retry has a 5s turn around.
+        '''
+        total = 5
+        found_match = False
+
+        for attempt in range(total):
+            try:
+                child.sendline(command)
+                child.expect(pattern, 5)
+                found_match = True
+                break
+            except pexpect.TIMEOUT:
+                logger.debug("Iteration '%s' out of '%s'", attempt, total)
+
+        if not found_match:
+            raise Exception("No '%s' in '%s'" % (pattern, command))
+
+    try:
+        child = pexpect.spawn("socat TCP:%s:%s -,raw,echo=0,escape=0x1d" % (localhost, CONSOLE_PORT))
+
+        if argv == logging.DEBUG:
+            child.logfile = sys.stdout
+
+        child.timeout = 600  # Long time for full configuration, waiting for ip address etc
+
+        # Setup username and password and log in
+        child.expect('Press RETURN to get started.', child.timeout)
+        child.sendline("")  # Send enter
+        child.expect('Enter root-system username:', child.timeout)
+        child.sendline("vagrant")
+        child.expect("Enter secret:")
+        child.sendline("vagrant")
+        child.expect("Enter secret again:")
+        child.sendline("vagrant")
+        child.expect("Username:")
+        child.sendline("vagrant")
+        child.expect("Password:")
+        child.sendline("vagrant")
+        child.expect(prompt)
+
+        # Set term width
+        child.sendline("term width 300")
+        child.expect(prompt)
+
+        # ZTP causes some startup issues so disable it during box creation
+        child.sendline("ztp terminate noprompt")
+        child.expect(prompt)
+
+        # Determine if the image is a crypto/k9 image or not
+        # This will be used to determine whether to configure ssh or not
+        child.sendline("bash -c rpm -qa | grep k9sec")
+        child.expect(prompt)
+        output = child.before
+        if '-k9sec' in output:
+            crypto = True
+            logger.debug("Crypto k9 image detected")
+        else:
+            crypto = False
+            logger.debug("Non crypto k9 image detected")
+
+        # Determine if the image has the MGBL package needed for gRPC
+        child.sendline("bash -c rpm -qa | grep mgbl")
+        child.expect(prompt)
+        output = child.before
+        if '-mgbl' in output:
+            mgbl = True
+            logger.debug("MGBL package detected")
+        else:
+            mgbl = False
+            logger.debug("MGBL package not detected")
+
+        # Wait for a management interface to be available
+        xr_cli_wait_for_output('sh run | inc MgmtEth', 'interface MgmtEth')
```

Comment:
Just a nit - 'sh run **interface** | inc MgmtEth' might be slightly more efficient.